### ISSUE TITLE
ValueError: Character [.] is not allowed in identifier after updating to SnowDDL 0.37.0

### ISSUE THREAD

**Issue by CM000n** [littleK0i/SnowDDL#163]
**Describe the bug**
After updating to version 0.37.0, I get the following error message when I try to run `snowddl plan`:
```
2024-12-05 13:36:43.731 - WARNING - Failed to parse [CDE_TEAM_MEMBER] in config file [/builds/customer-data-empowerment/infrastructure-and-deployment/snowddl/config/business_role.yaml]
Traceback (most recent call last):
  File "/builds/customer-data-empowerment/infrastructure-and-deployment/snowddl/venv/lib[64](https://gitlab.dm-drogeriemarkt.com/customer-data-empowerment/infrastructure-and-deployment/snowddl/-/jobs/47754706#L64)/python3.9/site-packages/snowddl/parser/abc_parser.py", line 66, in parse_multi_entity_file
    callback(entity_name, entity_params)
  File "/builds/customer-data-empowerment/infrastructure-and-deployment/snowddl/venv/lib64/python3.9/site-packages/snowddl/parser/business_role.py", line 123, in process_business_role
    global_roles=[Ident(global_role_name) for global_role_name in business_role_params.get("global_roles", [])],
  File "/builds/customer-data-empowerment/infrastructure-and-deployment/snowddl/venv/lib64/python3.9/site-packages/snowddl/parser/business_role.py", line 123, in <listcomp>
    global_roles=[Ident(global_role_name) for global_role_name in business_role_params.get("global_roles", [])],
  File "/builds/customer-data-empowerment/infrastructure-and-deployment/snowddl/venv/lib64/python3.9/site-packages/snowddl/blueprint/ident.py", line 84, in __init__
    self.name = self._validate_part(name)
  File "/builds/customer-data-empowerment/infrastructure-and-deployment/snowddl/venv/lib64/python3.9/site-packages/snowddl/blueprint/ident.py", line 54, in _validate_part
    raise ValueError(
ValueError: Character [.] is not allowed in identifier [SNOWFLAKE.OBJECT_VIEWER], only ASCII letters, digits and single underscores are accepted
2024-12-05 13:36:43.[74](https://gitlab.dm-drogeriemarkt.com/customer-data-empowerment/infrastructure-and-deployment/snowddl/-/jobs/47754706#L74)4 - ERROR - Execution halted due to [1] error(s) in config parsers
```

Please excuse me if I have missed something. But in the changelogs I have not found any reference to a change of the `global_roles` definition for `business_role` configurations.

**Expected behavior**
A `snowddl plan` should run without an error message

**Attach YAML config (if applicable)**
The related yaml config is this
```yaml
CDE_TEAM_MEMBER:
  comment: Default CDE team member role that can be applied to any cde team member.
  global_roles:
    - SNOWFLAKE.OBJECT_VIEWER
    - SNOWFLAKE.USAGE_VIEWER
    - SNOWFLAKE.GOVERNANCE_VIEWER
    - SNOWFLAKE.SECURITY_VIEWER
  database_owner:
    - PRIVATE
  database_write:
    - MSTR
  database_read:
    - MSTR
  warehouse_usage:
    - CDE_TEAM_MEMBER_WH
  warehouse_monitor:
    - SNOWDDL_WH
    - CDE_TEAM_MEMBER_WH
    - MSTR_COMPUTE_WH
    - MONITORING_WH

```

**Comment by littleK0i** [comment]
Yeah, it is mentioned in the main article: https://docs.snowddl.com/breaking-changes-log/0.37.0-december-2024

It is essentially the only "breaking" change in this version when using pure YAML configs. Simply rename `global_roles` to `share_read`, and it should work fine.

**Comment by CM000n** [comment]
Ahh, I didn't realize that. I thought it only affected the config for inbound shares and was not relevant for business roles. Thanks for the hint and the help 😊

**Comment by CM000n** [comment]
If I simply rename `global_roles` to `share_read`, I get the following error:

```
Traceback (most recent call last):
  File "/builds/customer-data-empowerment/infrastructure-and-deployment/snowddl/venv/bin/snowddl", line 8, in <module>
    sys.exit(entry_point())
  File "/builds/customer-data-empowerment/infrastructure-and-deployment/snowddl/venv/lib64/python3.9/site-packages/snowddl/app/base.py", line [71](https://gitlab.dm-drogeriemarkt.com/customer-data-empowerment/infrastructure-and-deployment/snowddl/-/jobs/47759405#L71)7, in entry_point
    app.execute()
  File "/builds/customer-data-empowerment/infrastructure-and-deployment/snowddl/venv/lib64/python3.9/site-packages/snowddl/app/base.py", line 592, in execute
    resolver.resolve()
  File "/builds/customer-data-empowerment/infrastructure-and-deployment/snowddl/venv/lib64/python3.9/site-packages/snowddl/resolver/abc_resolver.py", line 46, in resolve
    self.blueprints = self.get_blueprints()
  File "/builds/customer-data-empowerment/infrastructure-and-deployment/snowddl/venv/lib64/python3.9/site-packages/snowddl/resolver/business_role.py", line 10, in get_blueprints
    return {
  File "/builds/customer-data-empowerment/infrastructure-and-deployment/snowddl/venv/lib64/python3.9/site-packages/snowddl/resolver/business_role.py", line 11, in <dictcomp>
    full_name: self.transform_blueprint(business_role_bp)
  File "/builds/customer-data-empowerment/infrastructure-and-deployment/snowddl/venv/lib64/python3.9/site-packages/snowddl/resolver/business_role.py", line 20, in transform_blueprint
    grants.extend(self.build_database_role_grants(database_name_pattern, self.config.OWNER_ROLE_TYPE))
  File "/builds/customer-data-empowerment/infrastructure-and-deployment/snowddl/venv/lib64/python3.9/site-packages/snowddl/resolver/abc_role_resolver.py", line 387, in build_database_role_grants
    schema_name_pattern = IdentPattern(self.config.env_prefix, f"{database_bp.full_name.database}.*")
TypeError: __init__() takes 2 positional arguments but 3 were given
```

**Comment by littleK0i** [comment]
This is a bit odd. I see 2 arguments being passed, why does it complaint about 3.

```
 schema_name_pattern = IdentPattern(self.config.env_prefix, f"{database_bp.full_name.database}.*")
```

**Comment by littleK0i** [comment]
@CM000n , should be fixed in `0.37.1`. Oddly enough, this specific logic branch was not covered with test.

**Comment by CM000n** [comment]
Hi @littleK0i . After updating to `0.37.2` it's working again.
Thank you very much for your quick reply and fix ❤️ 😊
